src/test/java/net/snowflake/client/jdbc/MaxLobSizeIT.java

```diff
@@ -0,0 +1,371 @@
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+import java.nio.charset.Charset;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Random;
+
+public class MaxLobSizeIT extends BaseJDBCTest {
+
+    protected static String jsonQueryResultFormat = "json";
+
+    //TODO: increase max size to 128 * 1024 * 1024
+    public static int maxLobSize = 16 * 1024 * 1024;
+    public static int largeLobSize = maxLobSize/2;
+    public static int mediumLobSize = largeLobSize/2;
+    public static int originLobSize = mediumLobSize/2;
+    public static int smallLobSize = 16;
+
+    public static String enableMaxLobSize = "alter session set FEATURE_INCREASED_MAX_LOB_SIZE_IN_MEMORY = 'ENABLED'";
+    public static String insertQuery = "insert into my_lob_test (c1, c2, c3) values ('";
+    public static String preparedInsertQuery = "insert into my_lob_test (c1, c2, c3) values (?, ?, ?)";
+    public static String selectQuery = "select * from my_lob_test";
+
+    public static String generateRandomString(int stringSize) {
+        RandomStringGenerator randomStringGenerator = new RandomStringGenerator.Builder().withinRange('a','z').build();
+        return randomStringGenerator.generate(stringSize);
+    }
+
+    public static Connection getConnection() throws SQLException {
+        Connection con = BaseJDBCTest.getConnection();
+        Statement stmt = con.createStatement();
+//        stmt.execute(enableMaxLobSize);
+        stmt.close();
+        return con;
+    }
+
+    public static void setJSONResultFormat(Connection con) throws SQLException {
+        Statement stmt = con.createStatement();
+        stmt.execute("alter session set jdbc_query_result_format = '" + jsonQueryResultFormat + "'");
+        stmt.close();
+    }
+
+    public void createTable(int lobSize, Connection con) throws SQLException {
+        String createTableQuery = "create or replace table my_lob_test (c1 varchar, c2 varchar("
+                + lobSize
+                + "), c3 int)";
+
+        Statement stmt = con.createStatement();
+        stmt.execute(createTableQuery);
+        stmt.close();
+    }
+
+    public void insertQuery(String varCharValue, int intValue, Connection con) throws SQLException {
+        Statement stmt = con.createStatement();
+        stmt.executeUpdate(insertQuery + varCharValue + "', '" + varCharValue + "', " + intValue + ")");
+        stmt.close();
+    }
+
+    public void preparedInsertQuery(String varCharValue, int intValue, Connection con) throws SQLException {
+        PreparedStatement pstmt = con.prepareStatement(preparedInsertQuery);
+
+        pstmt.setString(1, varCharValue);
+        pstmt.setString(2, varCharValue);
+        pstmt.setInt(3, intValue);
+
+        pstmt.execute();
+        pstmt.close();
+    }
+    @Test
+    public void testSmallLobSize() throws SQLException {
```

Comment:
let's make those tests parameterized with the size of the lob and just comment out or ignore not supported yet entries